core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -246,6 +251,38 @@ class BlockManagerMasterEndpoint(
     blockManagerIdByExecutor.get(execId).foreach(removeBlockManager)
   }
 
+  private def recoverLatestRDDBlock(
+      execId: String,
+      excludeExecutors: Seq[String],
+      context: RpcCallContext): Unit = {
+    logDebug(s"Replicating first cached block on $execId")
+    val excluded = excludeExecutors.flatMap(blockManagerIdByExecutor.get)
+    val response: Option[Future[Boolean]] = for {
+      blockManagerId <- blockManagerIdByExecutor.get(execId)
+      info <- blockManagerInfo.get(blockManagerId)
+      blocks = info.cachedBlocks.collect { case r: RDDBlockId => r }
+      // we assume blocks from the latest rdd are most relevant
```

Comment:
On first read I thought you were _only_ replicating the latest rdd.  Maybe expand this to

As a heuristic, prioritize replicating the latest rdd.  If this succeeds, CacheRecoveryManager will try to replicate the remaining rdds.

Reply:
Hey @squito thanks again for the comments, sorry it took me a minute to get back to you. 

So, what you describe is basically what I'm doing. I start replicating the latest block and work back to the earliest blocks. If there are any problems I give up. I use the rdd block id as a proxy for the order in which blocks were created.

Reply:
right, I'm suggesting that you expand the *comment* to be what I had written above, so its a little easier to follow the logic.

Reply:
Oh I see, sorry. This is updated.